hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/JobSubmissionFiles.java

```diff
@@ -160,6 +160,7 @@ public static Path getStagingDir(Cluster cluster, Configuration conf,
       }
     } catch (FileNotFoundException e) {
       fs.mkdirs(stagingArea, new FsPermission(JOB_DIR_PERMISSION));
```

Comment:
I suggest using static method `FileSystem.mkdirs(fs, stagingArea, new FsPermission(JOB_DIR_PERMISSION))`. It's one less line of code and potentially one less remote call for certain `FileSystem` implementations.